### DISCUSSION TITLE
Riverpod in large scale applications

### DISCUSSION THREAD

**Discussion by AlexMichels** [rrousselGit/riverpod#3417] [Q&A] [answered]
Hello @rrousselGit ,
I really enjoy using Riverpod. Currently, I am using Riverpod in projects with over 100k lines of code and for hundreds of thousands of users. We are using Riverpod annotation, Feature First, and each feature consists of UI -> controller (Notifier) -> service. The Riverpod providers are the glue between our layers. The problem we have on a large scale is that the build runner takes 3-5 minutes to run. That means if I change the branch or work on a provider, it takes 4 minutes until it is done. This costs each engineer about 30 minutes per day.

What is your recommendation? I was thinking maybe to push the generated files so at least we do not need to wait so long each time we change the branch.

Are there better solutions to improve the use of annotation for large projects?

Thanks in advance.

**Comment by snapsl** [comment, marked as answer]
> What is your recommendation? I was thinking maybe to push the generated files so at least we do not need to wait so long each time we change the branch.

Yes, I would recommend adding generated files to the repo for this and other reasons.
If that is not enough, `build_runner` also has other options to optimize build time that are documented (e.g. using `build.yaml`). However, some of these optimization can lead to errors due to project specific dependencies. Therefore I cannot give a general answer here.

> Are there better solutions to improve the use of annotation for large projects?

For really big projects you could structure your features into  packages.
